agent_build_refactored/scripts/filter_matrix_jobs.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2014-2022 Scalyr Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This script has to be executed by CI/CD in order to filter out job matrices and remove jobs, that should not be
+executed if it's not a "master" run, aka run from master branch, pull request to master, or tag.
+"""
+
+import argparse
+import json
+import os
+import sys
+import pathlib as pl
+
+# This file can be executed as script. Add source root to the PYTHONPATH in order to be able to import
+# local packages. All such imports also have to be done after that.
+sys.path.append(str(pl.Path(__file__).parent.parent.parent))
+
+from agent_build_refactored import ALL_USED_BUILDERS
+
+DEFAULT_OS = os.environ["DEFAULT_OS"]
+DEFAULT_PYTHON_VERSION = os.environ["DEFAULT_PYTHON_VERSION"]
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--is-master-run",
+        required=True
+    )
```

Comment:
Since we get input value for this option from JSON, then it can be `true` or `false`, in case of `false` the `store_true` option will still be true, so, on the "command caller" side, we must not provide the `--is-master-run`  at all, but that's also adds some extra verbosity.

I guess we can do that with bash's conditional substitution, (like `${value:--is-master-run}`) but in this case we still have to return empty string instead of `false` in our JSON object.